src/Controller/BatchProcessingController.py

```diff
@@ -158,106 +168,155 @@ def perform_processes(self, interrupt_flag, progress_callback):
-                            PatientDictContainer().filepaths['rtss'],
-                            rtss.SOPInstanceUID,
-                            rtss.SOPClassUID,
-                            rtss.Modality)
-                        rtss_series.add_image(rtss_image)
-
-                        # Add the new study to the patient
-                        patient.studies[rtss.StudyInstanceUID].add_series(
-                            rtss_series)
-
-                        # Update the patient dict container
-                        PatientDictContainer().set("rtss_modified", False)
-                    reason = "SUCCESS"
-                else:
-                    reason = process.summary
-
-                # Append process summary
-                if patient not in self.batch_summary.keys():
-                    self.batch_summary[patient] = {}
-                self.batch_summary[patient]["suv2roi"] = reason
-                progress_callback.emit(("Completed SUV2ROI", 100))
+            # Perform processes on patient
+            for process in self.processes:
+                self.process_functions[process](interrupt_flag,
+                                                progress_callback,
+                                                patient)
 
         PatientDictContainer().clear()
 
+    def update_rtss(self, patient):
+        """
+        Updates the patient dict container with the newly created RTSS (if a
+        process generates one), so it can be used by future processes.
+        :param patient: The patient with the newly-created RTSS.
+        """
+        # Get new RTSS
+        rtss = PatientDictContainer().dataset['rtss']
+
+        # Create a series and image from the RTSS
+        rtss_series = Series(rtss.SeriesInstanceUID)
+        rtss_series.series_description = rtss.get(
+            "SeriesDescription")
+        rtss_image = Image(
+            PatientDictContainer().filepaths['rtss'],
+            rtss.SOPInstanceUID,
+            rtss.SOPClassUID,
+            rtss.Modality)
+        rtss_series.add_image(rtss_image)
+
+        # Add the new study to the patient
+        patient.studies[rtss.StudyInstanceUID].add_series(
+            rtss_series)
+
+        # Update the patient dict container
+        PatientDictContainer().set("rtss_modified", False)
+
+    def batch_iso2roi_handler(self, interrupt_flag,
+                              progress_callback, patient):
+        """
+        Handles creating, starting, and processing the results of batch
+        ISO2ROI.
+        :param interrupt_flag: A threading.Event() object that tells the
+                               function to stop loading.
+        :param progress_callback: A signal that receives the current
+                                  progress of the loading.
+        :param patient: The patient to perform this process on.
+        """
+        # Get current patient files
+        cur_patient_files = self.get_patient_files(patient)
```

Comment:
get_patient_files is a static method so It shouldn't be called by an instance. As a static member holds the state of the class, it's not a good practice that one instance can modify the state of other instances.

consider using your Classname to call get_patient_files

Reply:
These have been fixed!